dhis-2/dhis-services/dhis-service-core/src/test/java/org/hisp/dhis/dataelement/DataElementWithValueTypeOptionsTest.java

```diff
@@ -0,0 +1,104 @@
+
+/*
+ * Copyright (c) 2004-2020, University of Oslo
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import org.hisp.dhis.DhisSpringTest;
+import org.hisp.dhis.analytics.AggregationType;
+import org.hisp.dhis.category.CategoryCombo;
+import org.hisp.dhis.common.FileTypeValueOptions;
+import org.hisp.dhis.common.ValueType;
+import org.hisp.dhis.common.ValueTypeOptions;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+public class DataElementWithValueTypeOptionsTest extends DhisSpringTest
+{
+    @Autowired
+    private DataElementStore dataElementStore;
+
+    @Test
+    public void testDeleteAndGetDataElement()
+    {
+        DataElement dataElementA = createDataElementWithValueTypeOptions( 'A', null );
+
+        dataElementStore.save( dataElementA );
+
+        long idA = dataElementA.getId();
+
+        assertNotNull( dataElementStore.get( idA ) );
+
+        dataElementA = dataElementStore.get( idA );
+
+        ValueTypeOptions valueTypeOptions = dataElementA.getValueTypeOptions();
+
+        assertNotNull( valueTypeOptions );
+        assertEquals( FileTypeValueOptions.class, valueTypeOptions.getClass() );
+        assertEquals( 100L, ((FileTypeValueOptions) valueTypeOptions).getMaxFileSize() );
+
+        dataElementStore.delete( dataElementA );
+
+        assertNull( dataElementStore.get( idA ) );
```

Comment:
I find this formatting with many blank lines really taxing to read in a test because there is no clear organisation in steps like: setup, execution, verification.